src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -1460,9 +1472,61 @@ namespace xclcpuemhal2 {
       xclClose_RPC_CALL(xclClose,this);
 #endif
     }
+   closemMessengerThread();
    saveDeviceProcessOutput();
   }
 
+  void CpuemShim::closemMessengerThread()
+  {
+    if (mMessengerThreadStarted) {
+      mMessengerThread.join();
+      mMessengerThreadStarted = false;
+    }
```

Comment:
Will this not hang forever?   The thread loops on (inst) and I don't see it ever breaking out of the loop?

Reply:
We want the thread to be running forever (until emulation runs).Main intention being to parse and show some device process messages on the console. We are calling closemMessengerThread in the destructor and after xclClose call.